File: app/src/main/java/com/example/wsmodule2/MainJavaActivity.java
package com.example.wsmodule2;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.graphics.Bitmap;
import android.graphics.BitmapFactory;
import android.os.AsyncTask;
import android.os.Bundle;
import android.view.View;
import android.view.ViewGroup;
import android.widget.LinearLayout;
import android.widget.Space;
import android.widget.TextView;

import com.example.wsmodule2.POJO.AccountData;
import com.example.wsmodule2.POJO.CardData;
import com.example.wsmodule2.POJO.CreditData;
import com.example.wsmodule2.tasks.DeleteLogoutTask;

public class MainJavaActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        findViewById(R.id.main_logout).setOnClickListener((View v) -> {
            new DeleteLogoutTask().executeOnExecutor(AsyncTask.THREAD_POOL_EXECUTOR);
        });

        findViewById(R.id.main_profile).setOnClickListener((View v) ->{
            startActivity(new Intent(MainJavaActivity.this, ProfileActivity.class));
        });

        LinearLayout container = findViewById(R.id.main_container);
        LinearLayout.LayoutParams params = new LinearLayout.LayoutParams(
                ViewGroup.LayoutParams.MATCH_PARENT,
                40
        );

        TextView myCard = new TextView(getApplicationContext());
        myCard.setBackgroundColor(getResources().getColor(R.color.colorGray));
        myCard.setLayoutParams(params);
        myCard.setText("МОИ КАРТЫ");

        TextView myAccount = new TextView(getApplicationContext());
        myAccount.setBackgroundColor(getResources().getColor(R.color.colorGray));
        myAccount.setLayoutParams(params);
        myAccount.setText("МОИ СЧЕТА");

        TextView myCredit = new TextView(getApplicationContext());
        myCredit.setBackgroundColor(getResources().getColor(R.color.colorGray));
        myCredit.setLayoutParams(params);
        myCredit.setText("МОИ КРЕДИТЫ");

        Bitmap bmp = BitmapFactory.decodeResource(getResources(), R.drawable.matercard_96);
        LinearLayout.LayoutParams spaceParams = new LinearLayout.LayoutParams(
                ViewGroup.LayoutParams.MATCH_PARENT,
                10
        );

        container.addView(myCard);
        for(int i = 0; i < 3; i++)
        {
            CardData cardData = new CardData("22223144444", "Дебетовая карта "+i , bmp, 33.3333);
            AccountView accountView = new AccountView(getApplicationContext(),
                    cardData);
            accountView.setBackgroundColor(getResources().getColor(R.color.colorWhiteSmoke));
            accountView.setTag(cardData);
            container.addView(accountView);
            Space space = new Space(getApplicationContext());
            space.setLayoutParams(spaceParams);
            container.addView(space);
            AccountView accountView1 = container.findViewWithTag(cardData);
            StartActivity.cards.add(cardData);
            accountView1.setOnClickListener((View v) ->{
                Intent intent = new Intent(MainJavaActivity.this, CardsActivity.class);
                intent.putExtra("card_list_index", StartActivity.cards.indexOf(cardData));
                startActivity(intent);
            });
        }

        container.addView(myAccount);
        for(int i = 0; i<3; i++)
        {
            AccountView accountView = new AccountView(getApplicationContext(),
                    new AccountData("22223144444", "Дебетовая карта" , 33.3333));
            accountView.setBackgroundColor(getResources().getColor(R.color.colorWhiteSmoke));
            container.addView(accountView);
            Space space = new Space(getApplicationContext());
            space.setLayoutParams(spaceParams);
            container.addView(space);
        }

        container.addView(myCredit);
        for(int i = 0; i<3; i++)
        {
            AccountView accountView = new AccountView(getApplicationContext(),
                    new CreditData("Платеж аааа", "Дебетовая карта" , 33.3333));
            accountView.setBackgroundColor(getResources().getColor(R.color.colorWhiteSmoke));
            container.addView(accountView);
            Space space = new Space(getApplicationContext());
            space.setLayoutParams(spaceParams);
            container.addView(space);
        }
    }
}

File: app/src/main/java/com/example/wsmodule2/MainJavaActivity.kt
package com.example.wsmodule2

import android.content.Intent
import android.graphics.BitmapFactory
import android.os.AsyncTask
import android.os.Bundle
import android.util.Log
import android.view.View
import android.view.ViewGroup
import android.widget.LinearLayout
import android.widget.Space
import androidx.appcompat.app.AppCompatActivity
import com.addSpace
import com.example.wsmodule2.CustomViews.AccountView
import com.example.wsmodule2.POJO.AccountData
import com.example.wsmodule2.POJO.CardData
import com.example.wsmodule2.POJO.CreditData
import com.example.wsmodule2.StartActivity.Companion.accounts
import com.example.wsmodule2.StartActivity.Companion.cards
import com.example.wsmodule2.tasks.DeleteLogoutTask

class MainJavaActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        findViewById<View>(R.id.main_logout).setOnClickListener { v: View? -> DeleteLogoutTask().executeOnExecutor(AsyncTask.THREAD_POOL_EXECUTOR) }
        findViewById<View>(R.id.main_profile).setOnClickListener { v: View? -> startActivity(Intent(this@MainJavaActivity, ProfileActivity::class.java)) }
        val container = findViewById<LinearLayout>(R.id.main_container)
        val params = LinearLayout.LayoutParams(
                ViewGroup.LayoutParams.MATCH_PARENT,
                40
        )

        val bmp = BitmapFactory.decodeResource(resources, R.drawable.matercard_96)
        val spaceParams = LinearLayout.LayoutParams(
                ViewGroup.LayoutParams.MATCH_PARENT,
                10
        )

        val card_container = findViewById<LinearLayout>(R.id.main_cards_container)
        val accountContainer = findViewById<LinearLayout>(R.id.main_bills_container)
        val credit_container = findViewById<LinearLayout>(R.id.main_credits_container)

        //adding cards
        for (i in 0..2) {
            val cardData = CardData("22223144444", "Дебетовая карта $i", bmp, 33.3333)

            val accountView = AccountView(applicationContext,
                    cardData)
            accountView.setBackgroundColor(resources.getColor(R.color.colorWhiteSmoke))


            accountView.setOnClickListener { v: View? ->
                val intent = Intent(this@MainJavaActivity, CardsActivity::class.java)
                intent.putExtra("card_list_index", StartActivity.cards.indexOf(cardData))
                startActivity(intent)
            }

            card_container.addView(accountView)
            card_container.addSpace(10)

            cards.add(cardData)
        }

        //adding accounts
        for (i in 0..2) {
            val accountData = AccountData("22223144444", "Дебетовая карта", 33.3333)

            val accountView = AccountView(applicationContext,
                   accountData)
            accountView.setBackgroundColor(resources.getColor(R.color.colorWhiteSmoke))

            accountView.setOnClickListener { view ->
                val intent = Intent(this@MainJavaActivity, UserAccountsActivity::class.java)
                intent.putExtra("useraccount_list_index", accounts.indexOf(accountData))
                startActivity(intent)
            }

            accountContainer.addView(accountView)
            accountContainer.addSpace(10)

            accounts.add(accountData)
        }

        //adding credits
        for (i in 0..2) {
            val accountView = AccountView(applicationContext,
                    CreditData("Платеж аааа", "Дебетовая карта", 33.3333))
            accountView.setBackgroundColor(resources.getColor(R.color.colorWhiteSmoke))

            credit_container.addView(accountView)

            credit_container.addSpace(10)
        }

    }
}